### ISSUE TITLE
Scaling problems

### ISSUE THREAD

**Issue by squalsoft** [ravibpatel/AutoUpdater.NET#35]
On hi dpi displays it exists problems with scaling Autoupdater windows. They are very small because use old Winforms technology. It is better to change winforms to WPF windows.

**Comment by ravibpatel** [comment]
You are right but for now, you can handle update logic yourself and that way you can provide your own dialog.

**Comment by gmichaud** [comment]
I spent part of last night trying to find a solution to this problem and found a relatively simple fix - all hardcoded font sizes should be changed from pixels to points in the .resx files. The scaling works properly afterwards. I did it in this commit: https://github.com/gmichaud/AutoUpdater.NET/commit/e90d7a48a2b42bdfe719a3cd7e972e6416b5eb72

There are a few other issues; one is with the icons used. The scale mode could be changed to zoom to ensure scaling but a higher resolution image has to be used otherwise it will look pixelated. The other issue I found is related to the web browser control - it doesn’t scale automatically.

**Comment by gmichaud** [comment]
The solution does not appear to work for me. SetFont gets called *after* Windows Forms applies its scaling logic, and since fonts in the forms are already defined in pixels instead of points, they get scaled down: https://imgur.com/a/1jwZm

**Comment by ravibpatel** [comment]
Adding the following to Manifest file also make forms inside the library DPI-aware.

```xml
<application xmlns="urn:schemas-microsoft-com:asm.v3">
    <windowsSettings>
      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
    </windowsSettings>
  </application>
```

**Comment by gmichaud** [comment]
Did you see any impact of adding this manifest? This is initially what I tried doing (to completely turn off scaling) but it had no effect -- seems like it is controlled at process level and Excel is already doing it for us.

**Comment by ravibpatel** [comment]
Yes, It works wonders. Fonts look sharper like it works in WPF. See [before](https://imgur.com/gB3XWEX) and [after](https://imgur.com/X78f4Ab) images.

**Comment by gmichaud** [comment]
Where did you put the manifest? At .exe level or inside the library itself? I’m using AutoUpdater as part of an Excel add-in, and Excel itself is already configured for high DPI - this is what was causing the scaling issues in the first place for me.

**Comment by ravibpatel** [comment]
@gmichaud At .exe level. In the AutoUpdaterTest project.

**Comment by ravibpatel** [comment]
@gmichaud Can you try the latest commit. I added new option SetProcessDPIAwareness. Can you try it for both possibilities (true and false)? To make sure how scaling behaves for Excel add-in. For WPF this works automatically and if you are using Winforms manifest way is the best thing to do.

**Comment by gmichaud** [comment]
Sorry for the delay. I haven't tried to play with SetProcessDPIAwareness yet but I've had reports of issues causing the buttons at the bottom of the Update form to be hidden: https://imgur.com/a/XJpaS - I was able to reproduce this by setting my local resolution to 1024x768, 100% font size.

There appears to be some interaction with Visual Studio and what's the local resolution when the form is edited. I was able to see a similar behaviour in one instance just by opening the form in a lower resolution/font size.

I found some articles on StackOverflow but I don't fully understand the problem or what's the right solution:
https://stackoverflow.com/questions/14875325/why-is-my-form-being-resized-when-it-is-displayed
https://stackoverflow.com/questions/27633483/my-windows-form-keeps-on-shrinking-resizing-on-build

AutoScaleMode=Font appears to be what's causing the automatic resize of the form. After setting a MinimumSize I did get the result I want, but i'm not sure it is the right approach.
